pkg/occlient/occlient.go

```diff
@@ -1052,36 +1052,59 @@ func (c *Client) FollowBuildLog(buildName string) error {
 	return nil
 }
 
-// Delete calls oc delete
-// kind is always required (can be set to 'all')
-// name can be omitted if labels are set, in that case set name to ''
-// if you want to delete object just by its name set labels to nil
-func (c *Client) Delete(kind string, name string, labels map[string]string) (string, error) {
+// Delete takes labels as a input and based on it, deletes respective resource
+func (c *Client) Delete(labels map[string]string) error {
+	// convert labels to selector
+	selector := util.ConvertLabelsToSelector(labels)
 
-	args := []string{
-		"delete",
-		kind,
+	var errorList []string
+	// Delete DeploymentConfig
+	err := c.appsClient.DeploymentConfigs(c.namespace).DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		errorList = append(errorList, "unable to delete deploymentconfig")
+	}
+	// Delete Route
+	err = c.routeClient.Routes(c.namespace).DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		errorList = append(errorList, "unable to delete route")
+	}
+	// Delete BuildConfig
+	err = c.buildClient.BuildConfigs(c.namespace).DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		errorList = append(errorList, "unable to delete buildconfig")
 	}
+	// Delete Build
+	err = c.buildClient.Builds(c.namespace).DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: selector})
```

Comment:
We don't need to delete `Builds`. They should be deleted automatically by [garbage collection](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/)